### PR TITLE
cosmic-panel: unstable-2023-09-22 -> unstable-2023-11-13

### PR DESCRIPTION
**Pull request by nyabinary**
## Description of changes
Updating, formatting, moved cosmic-panel to new ``by-name`` package layout.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by alyssais — CHANGES_REQUESTED** [review]
Are there actual user-facing changes in this update?  We don't necessarily want to be updating to every new git rev of packages with committed Cargo.lock files, because they're very big and end up making the git history bigger forever.

File: pkgs/top-level/all-packages.nix
Comment (Nyabinary) [reply pkgs/top-level/all-packages.nix:30956]: Fixed that

**Comment by Nyabinary** [comment]
> Are there actual user-facing changes in this update? We don't necessarily want to be updating to every new git rev of packages with committed Cargo.lock files, because they're very big and end up making the git history bigger forever.

Theming support, so yes

**Comment by alyssais** [comment]
Can you either not reformat the file, not move the file, or have two separate commits?  As it is, the file has changed too much for git to realise it's been moved.

File: pkgs/by-name/co/cosmic-panel/package.nix
Comment (alyssais) [inline pkgs/by-name/co/cosmic-panel/package.nix:48]: I find this quite a bit less readable than having "--set", the key, and the value on one line.

Comment (Nyabinary) [reply pkgs/by-name/co/cosmic-panel/package.nix:48]: I formatted via nixpkgs-fmt

**Comment by Nyabinary** [comment]
> Can you either not reformat the file, not move the file, or have two separate commits? As it is, the file has changed too much for git to realise it's been moved.

done

File: pkgs/by-name/co/cosmic-panel/package.nix
Comment (alyssais) [reply pkgs/by-name/co/cosmic-panel/package.nix:48]: We have no official formatter, so we don't have to defer to the formatter when it formats code to be less readable than it was before.

Comment (Nyabinary) [reply pkgs/by-name/co/cosmic-panel/package.nix:48]: All other COSMIC packages have this type of formatting, so it's consistent to have it for this too.

Comment (alyssais) [reply pkgs/by-name/co/cosmic-panel/package.nix:48]: No they don't? cosmic-applets formats this exactly as it was before, and cosmic-comp is not formatted with nixpkgs-fmt either.

Comment (Nyabinary) [reply pkgs/by-name/co/cosmic-panel/package.nix:48]: cosmic-settings, and all other cosmic prs have this formatting applied.

Comment (alyssais) [reply pkgs/by-name/co/cosmic-panel/package.nix:48]: I respect that we have no full authoritative formatting rules, so I'm generally happy to let people format their own code however they want, but that's different to reformatting existing code to make it (IMO) less readable.

Comment (Nyabinary) [reply pkgs/by-name/co/cosmic-panel/package.nix:48]: @alyssais I mean I like it, and I'll probably maintain this fully, and it is more readable to me IMO, I'll most likely be the main maintainer since I'm kind of unofficially heading the COSMIC for NixOS effort? (There a NixOS matrix channel I made & mod in for COSMIC), another thing is I'll use this as my DE in NixOS, dogfooding the packages and finding issues etc.

Comment (Nyabinary) [reply pkgs/by-name/co/cosmic-panel/package.nix:48]: True so what exactly do you want me to amend?

Comment (alyssais) [reply pkgs/by-name/co/cosmic-panel/package.nix:48]: The only thing I really don't like about the reformatting, as I said, is that I think it's significantly less readable to have

```nix
    "--set"
    "bin-src"
    "target/${rust.lib.toRustTargetSpecShort stdenv.hostPlatform}/release/cosmic-panel"
```

than it is to have

```nix
    "--set" "bin-src" "target/${rust.lib.toRustTargetSpecShort stdenv.hostPlatform}/release/cosmic-panel"
```

**Review by alyssais — APPROVED** [review]

**Review by alyssais — CHANGES_REQUESTED** [review]
https://github.com/NixOS/nixpkgs/pull/266539#pullrequestreview-1725158875

**Comment by Nyabinary** [comment]
> [#266539 (review)](https://github.com/NixOS/nixpkgs/pull/266539#pullrequestreview-1725158875)

done

**Review by alyssais — APPROVED** [review]

**Review by alyssais — APPROVED** [review]

**Pull request merged**
